components/brave_federated/task/model_util.cc

```diff
@@ -0,0 +1,31 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/task/model_util.h"
+
+#include <cmath>
+
+namespace brave_federated {
+
+// TODO(lminto): What is this function actually computing? Mean cross entropy?
+float ComputeNLL(std::vector<float> true_labels,
```

Comment:
@stevelaskaridis what is this function actually computing? Couldn't we just call it compute **cross entropy**?

Reply:
Negative Log Likelihood. You can check [this](https://pytorch.org/docs/stable/generated/torch.nn.NLLLoss.html) for more.